test/addons-napi/test_array/test.js

```diff
@@ -19,19 +19,30 @@ const array = [
   ]
 ];
 
-assert.strictEqual(test_array.Test(array, array.length + 1),
-                   'Index out of bound!');
+assert.throws(
+  () => {
+    test_array.TestGetElement(array, array.length + 1);
+  },
+  /Index out of bounds!/
+);
 
 assert.throws(
   () => {
-    test_array.Test(array, -2);
+    test_array.TestGetElement(array, -2);
   },
   /Invalid index\. Expects a positive integer\./
 );
 
 array.forEach(function(element, index) {
-  assert.strictEqual(test_array.Test(array, index), element);
+  assert.strictEqual(test_array.TestGetElement(array, index), element);
 });
 
 
 assert.deepStrictEqual(test_array.New(array), array);
+
+assert(test_array.TestHasElement(array, 0));
+assert.strictEqual(false, test_array.TestHasElement(array, array.length + 1));
+
+assert(test_array.NewWithLength(0) instanceof Array);
+assert(test_array.NewWithLength(1) instanceof Array);
+assert(test_array.NewWithLength(2 ^ 32 - 1) instanceof Array);
```

Comment:
hmm, maybe coding at 2am in your hotel room is not always the best idea.  Will fix.